### ISSUE TITLE
Cannot compile: error: unexpected ident '__js_peel

### ISSUE THREAD

**Issue by gazben** [unrust/uni-gl#5]
Hi,

When I try to compile the unrust package this error comes up:
```
➜  unrust git:(master) rustc --version
rustc 1.33.0-nightly (ddab10a69 2018-12-23)
➜  unrust git:(master) cargo web start --example boxes --release
   Compiling uni-gl v0.1.0
   Compiling uni-pad v0.1.0 (/home/gazben/projects/unrust/uni-pad)
   Compiling uni-snd v0.1.0
error: unexpected ident '__js_peel'

error: aborting due to previous error
error: Could not compile `uni-gl`.

To learn more, run the command again with --verbose.
error: build failed
```
Did I mess up somewhere?

**Comment by ryankaplan** [comment]
I have the same issue although when I use the nightly channel, I get a more detailed error...

```
error: unexpected ident '__js_peel'

error[E0425]: cannot find function `snippet` in this scope
   --> src/webgl.rs:137:21
    |
137 |            let value = js_raw!({
    |   _____________________^
    |  |_____________________|
    | ||
138 | ||             var ctx = Module.gl.get(@{a0});
139 | ||             return Module.gl.add(ctx.createBuffer());
140 | ||         }, [self.reference] );
    | ||                             ^
    | ||_____________________________|
    | |______________________________not found in this scope
    |                                in this macro invocation
    |
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

I believe the error `cannot find function snippet in this scope` is talking about the `snippet` function defined in the `__js_raw_asm` macro in `stdweb`...

```
#[doc(hidden)]
#[macro_export]
macro_rules! __js_raw_asm {
    ($code:expr, $($token:expr),*) => {{
        #[$crate::private::js_raw_attr]
        fn snippet() {
            call!( $code, $($token),* );
        }

        snippet( $($token as *const u8),* )
    }};

    ($code:expr) => { __js_raw_asm!( $code, ) };
}
```

It looks like this macro [was changed recently](https://github.com/koute/stdweb/commit/6da47b8ff53c6aefd2d770465389f87da6584440?diff=unified) in a way that - I'm guessing - means it can no longer be used outside of `stdweb`.

I am pretty new to rust, so I'm not sure about this. But `fn snippet` is declared with `#[$crate::private::js_raw_attr]`. Maybe that behaves differently than `$crate::private::emscripten_asm_const_int` which is what it used before.

I'm not sure what the right way is to fix this. Although I'd be happy to take a stab at it if anyone could provide some guidance! (cc @jice-nospam, @edwin0cheng )

**Comment by edwin0cheng** [comment]
Sorry for late reply. I am busy recently.
The __js_raw_asm  macro is a hack for old stdweb, which speedup  js - rust unboxing. 
And if it is the problem right now, we can just use the normal js! macro.

**Comment by edwin0cheng** [comment]
@gazben And could you mind to tell me which cargo-web version you use ?

**Comment by ryankaplan** [comment]
@edwin0cheng ah, great! Thanks for the fast reply!

I was able to reproduce the issue as well and I'm on `cargo-web 0.6.23`. Just to be sure, here's the command I ran...

```
$ cargo web --version
cargo-web 0.6.23
```

**Comment by ryankaplan** [comment]
@edwin0cheng @jice-nospam I created a PR for the fix you suggested because it was small and easy. If you find something better that is fast and works with the new `stdweb`, feel free to reject/close it.

Link to PR: https://github.com/unrust/uni-gl/pull/6

**Comment by gazben** [comment]
@edwin0cheng Little late to the party, sorry about that. The version is: `cargo-web 0.6.23`

**Comment by jice-nospam** [comment]
I have no issue with uni-gl on windows with 
$ rustc --version
rustc 1.33.0-nightly (a2b0f247b 2018-12-30)

I have an issue with uni-glsl though :
```
   Compiling uni-glsl v0.1.0 (https://github.com/unrust/uni-glsl#a418f73a)
error[E0658]: The attribute `rustfmt_skip` is currently unknown to the compiler and may have meaning added to it in the future (see issue #29642)
   --> h:\jice\cargo\git\checkouts\uni-glsl-290da6b884e28c33\a418f73\src\preprocessor.rs:470:3
    |
470 | #[rustfmt_skip]
    |   ^^^^^^^^^^^^
```

With nightly rust, things tends to break sometimes. Going back to an older version or upgrading to the latest can help. I'll give it a try with the latest nightly...

**[update]** apparently, the new syntax is `#[rustfmt::skip]`

**Comment by jice-nospam** [comment]
my bad, I wasn't up to date with upstream

**Comment by jice-nospam** [comment]
so upstream version compiles fine for me :
$ rustc --version
rustc 1.33.0-nightly (d22fa2d87 2019-01-08)
$ cargo web --version
cargo-web 0.6.23

**Comment by juchiast** [comment]
A quick fix would be to stay with stdweb 0.4.10 since it's the latest version that works:
```
[target.wasm32-unknown-unknown.dependencies]
stdweb =  "=0.4.10"
```

**Comment by edwin0cheng** [comment]
Should be closed by #6 , if still don't works, i will reopen it.

**Comment by juchiast** [comment]
It works! Can you publish a new version to crates.io?

**Comment by jice-nospam** [comment]
@juchiast Published as 0.1.1.
